champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/sword/Sever.java

```diff
@@ -73,72 +91,73 @@ public SkillType getType() {
         return SkillType.SWORD;
     }
 
-    @EventHandler
-    public void onEntityInteract(PlayerInteractEntityEvent event) {
-        if (event.getHand() == EquipmentSlot.OFF_HAND) return;
-        rightClicked.put(event.getPlayer(), true);
-        if (event.getRightClicked() instanceof LivingEntity entity) {
-            onInteract(event.getPlayer(), entity);
-        } else {
-            onInteract(event.getPlayer(), null);
-        }
-        event.setCancelled(true);
-    }
-
-    @EventHandler
-    public void onInteract(PlayerInteractEvent event) {
-        if (event.getHand() == EquipmentSlot.OFF_HAND || !event.getAction().isRightClick()) return;
-        if (!rightClicked.getOrDefault(event.getPlayer(), false)) { // This means onInteract wasn't called through onEntityInteract
-            onInteract(event.getPlayer(), null);
-        }
-        rightClicked.remove(event.getPlayer()); // Reset the flag for next interactions
+    @Override
+    public Action[] getActions() {
+        return SkillActions.RIGHT_CLICK;
     }
 
-    private void onInteract(Player player, LivingEntity ent) {
-        if (!isHolding(player)) return;
-
-        int level = getLevel(player);
+    @Override
+    public void activate(Player player, int level) {
         if (level <= 0) {
             return;
         }
 
-        final PlayerUseSkillEvent event = UtilServer.callEvent(new PlayerUseSkillEvent(player, this, level));
-        if (event.isCancelled()) {
-            return;
-        }
+        Vector directionVector = player.getLocation().getDirection().normalize().multiply(hitDistance / 2);
+        Location midpointLocation = player.getLocation().clone().add(directionVector);
+        Location playerChestLocation = player.getLocation().clone().add(0, 1, 0);
 
-        if (ent == null) {
-            player.getWorld().playSound(player.getLocation(), Sound.ENTITY_SPIDER_HURT, 1.0F, 1.5F);
-            UtilMessage.simpleMessage(player, getClassType().getName(), "You failed <green>%s", getName());
-            return;
-        }
+        drawParticleLine(playerChestLocation, directionVector, hitDistance, player);
 
-        boolean withinRange = UtilMath.offset(player, ent) <= hitDistance;
-        boolean isFriendly = false;
-        if (ent instanceof Player damagee) {
-            isFriendly = UtilEntity.getRelation(player, damagee) == EntityProperty.FRIENDLY;
-        }
+        // use this to get make sure the entities are enemies
+        List<KeyValue<LivingEntity, EntityProperty>> keyValueList = UtilEntity.getNearbyEntities(player, midpointLocation, hitDistance / 2, EntityProperty.ENEMY);
```

Comment:
you can use getNearbyEnemies instead, removes the need for the EntityPropery.ENEMY

Reply:
But I want it to effect entities too?